### DISCUSSION TITLE
Conditional modules

### DISCUSSION THREAD

**Discussion by vulns-are-features-too** [polybar/polybar#2852] [Ideas]
I'd like to have modules that are only shown under certain conditions.

Take for example, a `wifi` module and an `ethernet` module. I want `wifi` to be shown if and only if I'm connected to only a wireless interface and no wired interface. This is because I don't care about `wifi` when I either (1) have cable ethernet plugged in or (2) am using a VM which doesn't have wifi to begin with.

Currently, I can just the bar to show just `wifi` on my laptop host and just `ethernet` when I'm in a VM, but I'd like to be able to have a single config that I don't need to edit per machine.

**Comment by patrick96** [comment]
Not showing the wifi module when there is no wifi in the VM should be easy, simply set `format-disconnected = ""`.

Not showwing the wifi module if an ethernet cable is plugged in is outside the scope for polybar.

You can manually hide modules using the `module_show` and `module_hide` [module actions(https://polybar.readthedocs.io/en/stable/user/actions.html#all-modules) with `polybar-msg` and you could automate that to run the right command when the network state changes (I think both systemd and NetworkManager have facilities for that).
